### PR TITLE
docs(features): update `upload.mdx`

### LINKED ISSUE #1494: Create Upload guide

### PR DESCRIPTION
**Pull request by dreyfus92**
#### What kind of changes does this PR include?

- New or updated content

#### Description

- Closes #1494 
- What does this PR change? Give us a brief description. 
Updates `/features/upload` page

### CODE REVIEW

**Comment by FabianLars** [comment]
ah damn, that's my bad. @amrbashir We still want to merge it into the http plugin right?

**Comment by amrbashir** [comment]
I think it is fine to keep as is (or removed) since a combination of `http + fs` plugins can achieve the same thing?

I am a bit hesitated to merge into http plugin because we will need to handle scope for the filesystem as well and also upload plugin have a bit more dependencies that may not concern most users using http plugin.

**Comment by FabianLars** [comment]
> a combination of http + fs plugins can achieve the same thing?

That'd be much slower because the data would have to go through the ipc. The v2 ipc may be faster, but not _that_ much faster :/

> also upload plugin have a bit more dependencies that may not concern most users using http plugin.

imo feature flags would be fine for that 🤷 

> because we will need to handle scope for the filesystem as well

Hmm, that's indeed annoying and a good reason...

**Comment by amrbashir** [comment]
I guess better to be kept as a standalone plugin

**Comment by FabianLars** [comment]
should we rename it to http-extra? 😂 (not actually meant as a joke, i don't like the upload name when it can download too)

**Comment by amrbashir** [comment]
sure, whatever sounds right with you

**Comment by dreyfus92** [comment]
@FabianLars, so should I close this?

**Comment by FabianLars** [comment]
No, seems like we keep the plugin. If we indeed rename it then we can just change the docs 🤷

File: src/content/docs/features/upload.mdx [hunk 1]
```diff
@@ -3,12 +3,87 @@ title: Upload
...
 
-<Stub>
-	Based on
-	https://github.com/tauri-apps/plugins-workspace/tree/v2/plugins/upload
-</Stub>
+Upload files from disk to a remote server over HTTP. Download files from a remote HTTP server to disk.
+
+## Setup
+
+_This plugin requires a Rust version of at least **1.64**_
```
Comment (vasfvitor) [inline src/content/docs/features/upload.mdx]: 1.75 (we need to find a better way to specify the plugin version)

File: src/content/docs/features/upload.mdx [hunk 2]
```diff
@@ -3,12 +3,87 @@ title: Upload
...
+    pnpm="pnpm tauri add upload"
+    cargo="cargo tauri add upload"
+    />
+
+    </TabItem>
+    <TabItem label="Manual">
+
+    	<Steps>
+
+    	1. Run `cargo add tauri-plugin-upload` to add the plugin to the project's dependencies in `Cargo.toml`.
```
Comment (vasfvitor) [inline src/content/docs/features/upload.mdx:35]: I liked the steps, now we should update all the other guides to match this.

**Review by vasfvitor — COMMENTED** [review]
I think it's ok, have you used the plugin yourself to make sure it works as intended and this code is reproducible?

File: src/content/docs/features/upload.mdx (hunk 1 above)
Comment (dreyfus92) [reply src/content/docs/features/upload.mdx]: I was thinking about adding a note e.g.

:::note
This plugin requires a Rust version of at least **1.64**
:::

File: src/content/docs/features/upload.mdx (hunk 2 above)
Comment (dreyfus92) [reply src/content/docs/features/upload.mdx:35]: yeah, I wanted to implement this one so everybody could see them in action. I liked the result tbh.

File: src/content/docs/features/upload.mdx (hunk 1 above)
Comment (vasfvitor) [reply src/content/docs/features/upload.mdx]: think it's too much, I may have expressed myself badly, I was referring to the fact that we would have to manually check and update versions

Comment (dreyfus92) [reply src/content/docs/features/upload.mdx]: If we could unify the Rust version of all the plugins would be ideal tho.

Comment (vasfvitor) [reply src/content/docs/features/upload.mdx]: https://github.com/tauri-apps/plugins-workspace/blob/v2/README.md?plain=1#L34

So it seems they all share the same Rust version

Comment (vasfvitor) [inline src/content/docs/features/upload.mdx]:
```suggestion
_This plugin requires a Rust version of at least **1.75**_
```

**Review by vasfvitor — APPROVED** [review]

File: src/content/docs/features/upload.mdx (hunk 1 above)
Comment (FabianLars) [reply src/content/docs/features/upload.mdx]: At least currently they do, yes. I just added it to all readmes so it's displayed correctly on crates.io/npm. If you think it's valuable we could keep them always matched (so raise msrv for all, if required by one), and/or matching it with tauri too.

**Pull request merged**